internal/tasks/marketplacesvm-push-disk-images-task/marketplaces-push-disk-images-task.yaml

```diff
@@ -0,0 +1,151 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: marketplaces-push-disk-images
+  labels:
+    app.kubernetes.io/version: "0.2.2"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task to push disk images to Cloud Marketplaces
+  params:
+    - name: snapshot_json
+      type: string
+      description: String containing a JSON representation of the snapshot spec
+    - name: concurrentLimit
+      type: string
+      description: The maximum number of images to be pulled at once
+      default: 3
+    - name: cloudMarketplacesSecret
+      type: string
+      description: Env specific secret containing the marketplaces credentials
+    - name: starmapMapping
+      type: string
+      description: The mappings for images to marketplaces using the StArMap V2 format 
+  results:
+    - name: result
+      description: Success if the task succeeds, the error otherwise
+  steps:
+    - name: pull-and-push-images-to-marketplaces
+      image: quay.io/konflux-ci/release-service-utils:7f7a156a835c773bbcd7e5d7e44df2f573db14f2
+      env:
+        - name: CLOUD_CREDENTIALS
+          valueFrom:
+            secretKeyRef:
+              name: $(params.cloudMarketplacesSecret)
+              key: key
+      script: |
+        #!/usr/bin/env bash
+        set -ex
+
+        STDERR_FILE=/tmp/stderr.txt
+
+        exitfunc() {
+            local err=$1
+            local line=$2
+            local command="$3"
+            if [ "$err" -eq 0 ] ; then
+                echo -n "Success" > "$(results.result.path)"
+            else
+                echo "$0: ERROR '$command' failed at line $line - exited with status $err" \
+                  > "$(results.result.path)"
+                if [ -f "$STDERR_FILE" ] ; then
+                    tail -n 20 "$STDERR_FILE" >> "$(results.result.path)"
+                fi
+            fi
+            exit 0 # exit the script cleanly as there is no point in proceeding past an error or exit call
+        }
+        # due to set -e, this catches all EXIT and ERR calls and the task should never fail with nonzero exit code
+        trap 'exitfunc $? $LINENO "$BASH_COMMAND"' EXIT
+
+        # Setup required variables
+        set +x
+        STARMAP_MAPPING=$(params.starmapMapping)
+        STARMAP_MAPPING_FILE=/tmp/starmap.yaml
+        echo "$STARMAP_MAPPING" > "$STARMAP_MAPPING_FILE"
+        set -x
+
+        BASE_DIR="$(mktemp -d)"
+        DISK_IMGS_DIR="${BASE_DIR}/starmap/CLOUD_IMAGES"
+        mkdir -p "${DISK_IMGS_DIR}"
+
+
+        RUNNING_JOBS="\j" # Bash parameter for number of jobs currently running
+        NUM_COMPONENTS=$(jq '.components | length' <<< "$SNAPSHOT_JSON")
+
+        process_component() { # Expected argument is [component json]
```

Comment:
So this will pull the image and prepare the data for the push, right? And then the push will always happen all in one go? So there is no way to make that parallel?

Reply:
Yes, the `process_component` will be called in parallel to prepare the data and then the push will be executed in one go through the `marketplacesvm_push_wrapper`. However, the tooling is multi-thread so in the end the push execution will be done in parallel

Reply:
Any comments on this?

Reply:
Oh! I didn't notice this, it got lost in many threads.

@johnbieren WDYT? Should I update?

Reply:
ok, thanks!